packages/flutter/lib/src/material/list_tile.dart

```diff
@@ -385,91 +404,553 @@ class ListTile extends StatelessWidget {
+  }
+
+  RenderBox _subtitle;
+  RenderBox get subtitle => _subtitle;
+  set subtitle(RenderBox value) {
+    _subtitle = _updateChild(_subtitle, value, _ListTileSlot.subtitle);
+  }
+
+  RenderBox _trailing;
+  RenderBox get trailing => _trailing;
+  set trailing(RenderBox value) {
+    _trailing = _updateChild(_trailing, value, _ListTileSlot.trailing);
+  }
+
+  // The returned list is ordered for hit testing.
+  Iterable<RenderBox> get _children sync *{
+    if (leading != null)
+      yield leading;
+    if (title != null)
+      yield title;
+    if (subtitle != null)
+      yield subtitle;
+    if (trailing != null)
+      yield trailing;
+  }
+
+  bool get isDense => _isDense;
+  bool _isDense;
+  set isDense(bool value) {
+    if (_isDense == value)
+      return;
+    _isDense = value;
+    markNeedsLayout();
+  }
+
+  bool get isThreeLine => _isThreeLine;
+  bool _isThreeLine;
+  set isThreeLine(bool value) {
+    if (_isThreeLine == value)
+      return;
+    _isThreeLine = value;
+    markNeedsLayout();
+  }
+
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (_textDirection == value)
+      return;
+    _textDirection = value;
+    markNeedsLayout();
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    for (RenderBox child in _children)
+      child.attach(owner);
+  }
+
+  @override
+  void detach() {
+    super.detach();
+    for (RenderBox child in _children)
+      child.detach();
+  }
+
+  @override
+  void redepthChildren() {
+    _children.forEach(redepthChild);
+  }
+
+  @override
+  void visitChildren(RenderObjectVisitor visitor) {
+    _children.forEach(visitor);
+  }
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    final List<DiagnosticsNode> value = <DiagnosticsNode>[];
+    void add(RenderBox child, String name) {
+      if (child != null)
+        value.add(child.toDiagnosticsNode(name: name));
+    }
+    add(leading, 'leading');
+    add(title, 'title');
+    add(subtitle, 'subtitle');
+    add(trailing, 'trailing');
+    return value;
+  }
+
+  @override
+  bool get sizedByParent => false;
+
+  static double _minWidth(RenderBox box, double height) {
+    return box == null ? 0.0 : box.getMinIntrinsicWidth(height);
+  }
+
+  static double _maxWidth(RenderBox box, double height) {
+    return box == null ? 0.0 : box.getMaxIntrinsicWidth(height);
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    final double leadingWidth = _minWidth(leading, height);
+    return (leadingWidth <= 40.0 ? 56.0 : leadingWidth + 16.0)
+      + math.max(_minWidth(title, height), _minWidth(subtitle, height))
+      + _minWidth(trailing, height);
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    final double leadingWidth = _maxWidth(leading, height);
+    return (leadingWidth <= 40.0 ? 56.0 : leadingWidth + 16.0)
```

Comment:
Ditto here, and below.

Reply:
I haven't introduced one-time constants for the default tile heights and baselines because making up names for these values doesn't seems to make the code any more readable.

I did add a general comment about the dimensions and a pointer to the MD spec

Reply:
Yay!  We can add pointers to the MD spec again!